public/app/core/components/AppChrome/types.ts

```diff
@@ -5,3 +5,28 @@ export interface ToolbarUpdateProps {
   pageNav?: NavModelItem;
   actions?: React.ReactNode;
 }
+
+export interface HistoryEntryView {
+  name: string;
+  description: string;
+  url: string;
+  time: number;
+}
+
+export interface HistoryEntrySparkline {
+  values: number[];
+  range: {
+    min: number;
+    max: number;
+    delta: number;
+  };
+}
+
+export interface HistoryEntry {
+  name: string;
+  time: number;
+  breadcrumbs: NavModelItem[];
+  url: string;
+  views: HistoryEntryView[];
+  sparklineData?: HistoryEntrySparkline;
```

Comment:
I would expect some nesting here we can differentiate when you switch application (so collapse it in history), so you can clearly see in the history when you switch between different apps, and different app views